### PR TITLE
Document standalone usage of OSIAM

### LINKED ISSUE #69: Switch to a Spring Boot application with runnable WAR
Moved from https://github.com/osiam/resource-server/issues/43 & https://github.com/osiam/auth-server/issues/39

This is the umbrella issue for the transition of this component to a Spring Boot standalone application with an executable WAR. See also osiam/osiam#5 for the initial reasons and discussion.

Tasks that have to be done:
- [x] Change Hibernate naming strategy
- [x] Switch from XML to Java configuration
- [x] Introduce Spring Boot, but keep WAR deployment
- [x] Switch to YAML configuration (Issues: #70)
- [x] Load configuration and assets from the file system instead of the classpath
- [x] Extract assets and a working default configuration on startup
- [x] Run extraction of assets and configuration without actually start OSIAM
- [x] Document the standalone way of running OSIAM (Issue: #190)
- [x] Change ITs to use OSIAM as standalone application

Please, maintain this list, if you add new issues or close existing ones.

### LINKED ISSUE #190: Standalone Application Deployment in Production?
The standalone application deployment will be the new recommended way to run OSIAM. While running OSIAM with `java -jar` on the command line might be a viable way for testing or running in containers, but what about running in a production environment without containers? Shouldn't we supply some startup scripts, i.e. SysVinit scripts or systemd units? How is an admin supposed to easily run OSIAM in production? I think the ["detailed reference installation"](https://github.com/osiam/osiam/blob/master/docs/detailed-reference-installation.md) is supposed to be the guide about that, but what should we put there regarding standalone application deployment?

Part of #69

### PR DESCRIPTION
**Pull request by tkrille**
Resolves #69
Resolves #190

### CODE REVIEW

**Comment by tkrille** [comment]
I did not fully test this yet, but I've put it to review to catch all language mistakes as early as possible :)

File: docs/detailed-reference-installation.md [hunk 1]
```diff
@@ -54,6 +56,27 @@ OSIAM is distributed as a self-contained `.war` file.
...
+
+You should make the `.war` file executable now:
+
+```sh
+chmod +x osiam.war
+```
+
+**NOTE:** This is a feature of Spring Boot, the underlying framework OSIAM uses.
+It works by prepending a regular Bash script to the `.war` file.
+Most applications fill just ignore the script and process the `.war` part of the file.
```
Comment (wallner) [inline docs/detailed-reference-installation.md]: "Most applications fill"?

Comment (tkrille) [inline docs/detailed-reference-installation.md]: fixed

File: docs/detailed-reference-installation.md [hunk 2]
```diff
@@ -54,6 +56,27 @@ OSIAM is distributed as a self-contained `.war` file.
...
+It might work in a cygwin environment, but that has not been thoroughly tested.
+Essentially you need Bash, some standard UNIX commands and Java.
+
+You should make the `.war` file executable now:
+
+```sh
+chmod +x osiam.war
+```
+
+**NOTE:** This is a feature of Spring Boot, the underlying framework OSIAM uses.
```
Comment (wallner) [inline docs/detailed-reference-installation.md]: Maybe turn "Spring Boot" into a link to the homepage?

File: docs/detailed-reference-installation.md [hunk 3]
```diff
@@ -106,13 +129,137 @@ The configuration should be fully created before you can run the command.
...
 
 ## Starting OSIAM
 
+OSIAM can be started in two different ways: as a standalone application or as a Tomcat deployment.
+Both ways are equally well supported, but we recommend to use standalone application deployment.
+It's highly recommended to run OSIAM under a distinct service user.
+Never run OSIAM under the root user.
+
+**NOTE:** You have to find a directory, where you can place the `.war` file in.
+The simplest solution to that issue is to use OSIAM's home directory.
```
Comment (wallner) [inline docs/detailed-reference-installation.md]: "The simplest solution being"

**Comment by wallner** [comment]
I think it's fine.

**Comment by tkrille** [comment]
Updated, but I still have to thoroughly test this stuff.

**Comment by tkrille** [comment]
@wallner: I can confirm now, that this works. I made some small fixes though. It can be merged now.

**Comment by wallner** [comment]
Looks good. Thank you!

**Pull request merged**